Break up router and registry topics + add stand-alone registry install + Dan's prereqs reorg

adellape opened the pull request:
Split `install/docker_registry.adoc` into "Setting up the Registry" subdir
Still WIP: stand_alone_registry.adoc still placeholder. All xrefs haven't been updated.

Rebasing off https://github.com/openshift/openshift-docs/pull/2772 to pull in Dan's build_cfg changes and work off that.

@openshift/team-documentation The actual stand-alone registry install topic is currently still blank, but worth taking a look at would be what I've done to break up the enormous `docker_registry.adoc` topic into a subdirectory of multiple topics. Preview:

http://file.rdu.redhat.com/~adellape/090116/standalone_registry/install_config/registry/index.html

vikram-redhat commented:
@bfallonf @tpoitras @ahardin-rh - can you please review the registry topics breakup?

bfallonf commented:
This looks good from what I can tell. It's a bit hard to look at in its current state, but the TOC looks fine.

ahardin-rh commented:
I think previous conversation indicated that we are going to break Prerequisites up into separate System Requirements and Environment Requirements pages, but I don't see that reflected here.

bfallonf commented:
@ahardin-rh Someone else may be able to clarify, but I think the information inside the Prerequisites topic has just been rearranged to make more sense and more relevant instead of breaking it up into two topics, which I think is the better idea. At least, I think that was Dan's plan.

ahardin-rh commented:
@bfallonf Okay, thanks!

adellape commented:
@bfallonf @ahardin-rh @tpoitras @vikram-redhat @aheslin Still finishing up the "Installing a Stand-alone Registry" topic, but since my last comment I've rebased and also broken up the router topic into a subdirectory w/ separate topics (like I already had for the registry topic). Worth another perusal now, I think (though note I will fix up all the changed xrefs at the end before merging, if you come across busted ones):

http://file.rdu.redhat.com/~adellape/090116/standalone_registry/welcome/index.html

So to sum up again, at this point the overall changes are:

- Dan's prereqs reorg ("Planning" through "Containerized Components")
- Broke up the registry topic into the "Setting up the Registry" subdirectory
- Broke up the router topic into the "Setting up a Router" subdirectory
- Added "Installing a Cluster -> Installing a Stand-alone Registry" (WIP)

LMK if you see anything that needs adjusting. Thanks.

bfallonf commented:
@adellape Thumbs up from me. You've done some stuff I was just thinking needed to be done as well (organising the router shards topics).

ahardin-rh commented:
@adellape Looks great!

adellape commented:
Great, thanks @bfallonf @ahardin-rh!

adellape commented:
@aweiteka @smunilla @abutcher @ahardin-rh First draft of the stand-alone registry install doc done:

http://file.rdu.redhat.com/~adellape/090116/standalone_registry/install_config/install/stand_alone_registry.html#registry-advanced-installation

PTAL? Looking for tech and peer review before merging. Thanks.

ahardin-rh commented:
@adellape Looks good and accurate to me!

adellape commented:
@aweiteka @ahardin-rh Added the registry console info from https://github.com/openshift/openshift-docs/pull/2696 in a separate commit (will squash at the end):

http://file.rdu.redhat.com/~adellape/090116/standalone_registry/install_config/registry/deploy_registry_existing_clusters.html#registry-console

PTAL.

adellape commented on install_config/registry/deploy_registry_existing_clusters.adoc:
```diff
@@ -0,0 +1,450 @@
+====
+
+If no certificate is found, a self-signed certificate is created using the
+`*openssl` command and stored in the *_0-self-signed.cert_* file.
```
Oops!

adellape replied:
Fixed.

ahardin-rh commented:
@adellape just one comment from me. Thanks!

adellape commented:
Fixed links due to topic moves and some other broken ones I found.

@openshift/team-documentation Merging this big ol PR now. Probably will affect some open PRs, so be on the look out. Latest preview:

http://file.rdu.redhat.com/~adellape/090116/standalone_registry/install_config/index.html

php-coder commented on install_config/registry/deploy_registry_existing_clusters.adoc:
```diff
@@ -0,0 +1,447 @@
+$ sudo chown 1001:root <path>
+----
+====
+endif::[]
```
Asciibinder complains on this line:
```
  - install_config/registry/deploy_registry_existing_clusters
asciidoctor: ERROR: <stdin>: line 292: unmatched macro: endif::[]
```
